### PR TITLE
Update name attribute info

### LINKED ISSUE #24446: "Name" is no longer required for repeated elements in XML configuration
**Issue description**

The docs here ( https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#xml-configuration-provider ) mention that you need to use a Name attribute to have repeated elements in an appsettings.xml file. While the Name attribute will still work, the config will also work perfectly fine without it. (Since .NET 6 preview 4)
The only special case is when you only have one element but still want the "0" index in the key. Then you do need the Name attribute. In other cases, the XML config provider will automatically index all repeating elements.

See the pull request for more information: https://github.com/dotnet/runtime/pull/44608

**Target framework**

- [x] .NET Core 6 preview 4
- [ ] .NET Framework
- [ ] .NET Standard

[Edited below by gewarren]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d4c11507-cd06-292f-53bc-3a7f14affd86
* Version Independent ID: 33eb4890-44e0-522b-b1c1-964fec385b6a
* Content: [Configuration providers in .NET](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#xml-configuration-provider)
* Content Source: [docs/core/extensions/configuration-providers.md](https://github.com/dotnet/docs/blob/main/docs/core/extensions/configuration-providers.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #24446.

### CODE REVIEW

**Comment by gewarren** [comment]
cc @amoerie

File: docs/core/extensions/configuration-providers.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ An example *appsettings.xml* file with various configuration settings follows:
 
 :::code language="xml" source="snippets/configuration/console-xml/appsettings.xml":::
 
-Repeating elements that use the same element name work if the `name` attribute is used to distinguish the elements:
+In .NET 5 and earlier version, add the `name` attribute to distinguish repeating elements that use the same element name. In .NET 6 and later versions, the XML configuration provider automatically indexes repeating elements. That means you don't have to specify the `name` attribute, except if you want the "0" index in the key and there's only one element.
```
Comment (tdykstra) [inline docs/core/extensions/configuration-providers.md:91]:
```suggestion
In .NET 5 and earlier versions, add the `name` attribute to distinguish repeating elements that use the same element name. In .NET 6 and later versions, the XML configuration provider automatically indexes repeating elements. That means you don't have to specify the `name` attribute, except if you want the "0" index in the key and there's only one element.
```

**Review by tdykstra — APPROVED** [review]
LGTM

**Comment by gewarren** [comment]
Thanks Tom!

**Comment by amoerie** [comment]
Perfect, thanks!

**Pull request merged**